app/code/Magento/SalesRule/Model/Quote/Discount.php

```diff
@@ -88,19 +56,16 @@ class Discount extends AbstractTotal
      * @param RulesApplier|null $rulesApplier
      */
     public function __construct(
-        ManagerInterface $eventManager,
-        StoreManagerInterface $storeManager,
+        protected readonly ManagerInterface $eventManager,
+        protected readonly StoreManagerInterface $storeManager,
         Validator $validator,
```

Comment:
Why not promote this property?
```suggestion
        protected readonly Validator $calculator,
```